tests/by-util/test_install.rs

```diff
@@ -327,14 +327,17 @@ fn test_install_copy_file() {
 
 #[test]
 #[cfg(target_os = "linux")]
-#[ignore] // workaround from #1646 breaks this
 fn test_install_target_file_dev_null() {
     let (at, mut ucmd) = at_and_ucmd!();
     let file1 = "/dev/null";
     let file2 = "test_install_target_file_file_i2";
 
-    ucmd.arg(file1).arg(file2).succeeds().no_stderr();
-    assert!(at.file_exists(file2));
+    assert!(ucmd
+        .arg(file1)
+        .arg(file2)
+        .fails()
+        .stderr
+        .contains("cannot install")); // see #1646
```

Comment:
this must work
try:
install /dev/null foo.empty

Reply:
```
❯ ./target/debug/coreutils install /dev/null foo.empty
install: error: cannot install '/dev/null' to 'foo.empty': the source path is not an existing regular file
```

Reply:
it work on my system:
```
 ./target/debug/coreutils install /dev/null foo.txt   
```

Reply:
Hmmm...are you on `master`? Or, in any case, does your branch include 51383e10e65aa77c83c5789267977824e0ef9aee?

Reply:
yeah, and it includes my previous change :)
i just tried with 
45acb087b8764d20285f66098d70a9f1659140aa

Reply:
Rebased onto 45acb087b8764d20285f66098d70a9f1659140aa (`upstream/master` as of right now) and force-pushed. Still no luck; the `/dev/null` workaround remains there in `install.rs:491`:
```
❯ cargo test --features install --no-default-features |& grep null
test test_install::test_install_target_file_dev_null ... FAILED
---- test_install::test_install_target_file_dev_null stdout ----
run: /srv/projects/uutils/target/debug/coreutils install /dev/null target_file
thread 'test_install::test_install_target_file_dev_null' panicked at 'assertion failed: self.success', tests/common/util.rs:86:9
    test_install::test_install_target_file_dev_null
❯ ./target/debug/coreutils install /dev/null foo.empty   
install: error: cannot install '/dev/null' to 'foo.empty': the source path is not an existing regular file
❯ grep -A 6 dev.null src/uu/install/src/install.rs
    if from.to_string_lossy() == "/dev/null" {
        /* workaround a limitation of fs::copy
         * https://github.com/rust-lang/rust/issues/79390
         */
        if let Err(err) = File::create(to) {
            show_error!(
                "install: cannot install '{}' to '{}': {}",
❯ git log --oneline | head -2
6c0086a3 install: implement `-C` / `--compare`
45acb087 Merge pull request #1848 from nomius10/install_bugfix
```

Reply:
what is your os and rust version?

Reply:
```
❯ uname -mro && rustc --version
5.10.24_1 x86_64 GNU/Linux
rustc 1.45.2 (d3fb005a3 2020-07-31)
❯ rustup update
<...>
❯ rustc --version
rustc 1.50.0 (cb75ad5db 2021-02-10)
❯ cargo test --features install --no-default-features
<...>
thread 'test_install::test_install_target_file_dev_null' panicked at 'assertion failed: self.success', tests/common/util.rs:86:9
<...>
❯ ./target/debug/coreutils install /dev/null foo.empty
install: error: cannot install '/dev/null' to 'foo.empty': the source path is not an existing regular file
```
If it's [written into the source code](https://github.com/uutils/coreutils/blob/45acb087b8764d20285f66098d70a9f1659140aa/src/uu/install/src/install.rs#L490-L502) that copying /dev/null should fail, no version of Rust is going to change that.

Reply:
The code doesn't say that (I wrote it).

It is just an ugly code that will create an empty file if the source is /dev/null
The show_error is just if the File::create failed

Reply:
Aha, my mistake — I read it too quickly. In that case, I'm stumped. I'll come back to this in a bit.

Reply:
Hmm, if it works on everyone else's machine but mine, better to leave that test alone. Pardon the bother.